### ISSUE TITLE
Unable to install due to a Python version conflict

### ISSUE THREAD

**Issue by csolisr** [YunoHost-Apps/searx_ynh#19]
When attempting to install Searx, either from master or the `global_update` branch, I keep getting the following error:

    Downloading/unpacking cryptography>=1.3.4 (from pyopenssl==16.2.0->r /opt/yunohost/searx/requirements-ynh.txt (line 10))
      Running setup.py (path:/tmp/pip-build-H058kX/cryptography/setup.py) egg_info for package cryptography
        error in cryptography setup command: invalid environment marker: python_version < '3'

Apparently this is either an issue with `cryptography` wrongly attempting to use Python 3 instead of Python 2, or as another thread suggested a problem with `pip` and `setuptools`. I tried fixing it through the latter method, manually updating both packages through `pip`, but this resulted in the `yunohost` package ceasing to work entirely.

**Comment by csolisr** [comment]
Further notices: removing the version numbers from `https://github.com/YunoHost-Apps/searx_ynh/blob/master/sources/extra_files/app/requirements-ynh.txt` (thus using the latest versions of everything) does allow to compile and install the app.

**Comment by maniackcrudelis** [comment]
I tried to remove the versions numbers.
Unfortunately, it seems to work for the installation, but failed by the same way during an upgrade /o\

**Comment by maniackcrudelis** [comment]
After several tests, by removing the version number only for pyopenssl in requirements-ynh.txt

There no errors during the installation because pyopenssl and cryptography are keep as they are.
```
Requirement already satisfied (use --upgrade to upgrade): pyopenssl in /usr/lib/python2.7/dist-packages (from -r /opt/yunohost/searx/requirements-ynh.txt (line 32))
Requirement already satisfied (use --upgrade to upgrade): cryptography>=0.2.1 in /usr/lib/python2.7/dist-packages (from pyopenssl->-r /opt/yunohost/searx/requirements-ynh.txt (line 32))
```

But during the upgrade, pyopenssl is upgraded and there's the same error than you said.
```
Downloading/unpacking pyopenssl from https://pypi.python.org/packages/24/37/89bf12e53f1d27e8b2c8e5f8f9c7a958a3905f6916a9294a57a9d83fa165/pyOpenSSL-17.3.0-py2.py3-none-any.whl#md5=4217027ddfa4182d7f3b425d0b5eac92 (from -r /opt/yunohost/searx/requirements-ynh.txt (line 32))
Downloading/unpacking cryptography>=1.9 (from pyopenssl->-r /opt/yunohost/searx/requirements-ynh.txt (line 32))
```

**Comment by JimboJoe** [comment]
I've just commit a potential fix in this PR https://github.com/YunoHost-Apps/searx_ynh/pull/18

**Comment by JimboJoe** [comment]
Fix now merged in testing #20, along with 0.13.1 upstream version upgrade.
